app/models/merchant.rb

```diff
@@ -3,25 +3,24 @@ class Merchant < ActiveRecord::Base
   has_many :items
   has_many :invoice_items, through: :invoices
 
-  def self.revenue(params)
-    merchant = Merchant.find(params[:id])
+  def revenue(params)
     if params[:date]
-      rev = merchant.invoices.where(created_at: params[:date]).successful_transactions.joins(:invoice_items).sum("unit_price * quantity")
+      rev = invoices.where(created_at: params[:date]).successful_transactions.joins(:invoice_items).sum("unit_price * quantity")
     else
-      rev = merchant.invoices.successful_transactions.joins(:invoice_items).sum("unit_price * quantity")
+      rev = successful_invoices.joins(:invoice_items).sum("unit_price * quantity")
     end
-    revenue_format = (rev.to_f / 100).to_s
-    { revenue: revenue_format }
   end
 
-  def self.favorite_customer(id)
-    merchant = Merchant.find(id)
-    customer_id = merchant.invoices.successful_transactions.group_by(&:customer_id).max_by { |k, v| v.count }.flatten.first
+  def favorite_customer
+    customer_id = successful_invoices.group_by(&:customer_id).max_by { |k, v| v.count }.flatten.first
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Line is too long. [101/80]